### PR TITLE
Connect GraphQL to magazine related blocks #136

### LINKED ISSUE #136: Connect GraphQL to magazine related blocks
### Context/Description

All blocks in VT that currently fetch articles using the outdated magazine-articles.xlsx spreadsheet must be updated to utilise the GraphQL search endpoint.

Affected blocks:
- `related-articles`
- `v2-article-cards`
- `v2-stories-carousel`

The `magazine-articles` block already queries the new endpoint, but its logic needs to be updated and made more generic to enable reuse in other blocks.

### Acceptance Criteria

- [x] All blocks that retrieve the magazine article list must switch to the new database endpoint, discontinuing the use of the SharePoint magazine-articles.xlsx file.
- [x] Blocks that don’t require the entire article list should fetch only the necessary data by passing parameters (e.g., limit, truck, etc.).
- [x] The updated function for fetching magazine articles should reside in magazine.service.js, ensuring it’s accessible for future use. Any helper functions in this file must also be updated to ensure compatibility.
- [x] Ensure all affected blocks continue to function as before, but now retrieve data from the new GraphQL endpoint.
- [x] Ensure that every magazine article has the correct tags defined and updated. Instead of listing all tags under a single metadata property, `tags`, as before, the articles must now use three specific tags: `article-category`, `topic`, and `truck`.

### Notes for developers:
- Develop a function that accepts configurable parameters, allowing each block to fetch magazine articles as needed.
- Centralise some logic in magazine.service.js: the `magazine-articles` block already includes a function for fetching magazine articles. However, its logic should be moved to `magazine.service.js` and made generic so it can be reused by other blocks. This change requires removing the existing fetchMagazineArticles function and updating some helper functions currently in use.
- Eliminate unused logic from the blocks to streamline their functionality.
- Update the logic of the blocks as necessary to accommodate changes in the structure of the fetched data.
- [Documentation](https://projects.netcentric.biz/wiki/display/VOLVOFN/Search%3A+EDS+Search+Query).

### Notes for QAs:
- This is essentially a code refactoring. For testing, focus on ensuring that the last two AC are met. Be aware that the last AC might impact some blocks, and the articles displayed in certain blocks may differ slightly from what is currently shown. However, it is important to ensure the proper tags are defined everywhere first, and the pages will need to be republished.

### PR DESCRIPTION
**Pull request by amglebihan**
Fix #136

URL for testing:

Magazine Articles
- Before: https://main--volvotrucks-us--volvogroup.aem.page/block-library/blocks/magazine-articles
- After: https://136-connect-graphql-magazine-blocks--volvotrucks-us--volvogroup.aem.page/block-library/blocks/magazine-articles

Related articles (at the bottom of the page)
- Before: https://main--volvotrucks-us--volvogroup.aem.page/news-and-stories/volvo-trucks-magazine/25-years-of-driving-progress/
- After: https://136-connect-graphql-magazine-blocks--volvotrucks-us--volvogroup.aem.page/news-and-stories/volvo-trucks-magazine/25-years-of-driving-progress/

V2 Article Cards
- Before: https://main--volvotrucks-us--volvogroup.aem.page/block-library/blocks/v2-article-cards
- After: https://136-connect-graphql-magazine-blocks--volvotrucks-us--volvogroup.aem.page/block-library/blocks/v2-article-cards

V2 Stories Carousel
- Before: https://main--volvotrucks-us--volvogroup.aem.page/block-library/blocks/v2-stories-carousel
- After: https://136-connect-graphql-magazine-blocks--volvotrucks-us--volvogroup.aem.page/block-library/blocks/v2-stories-carousel

### CODE REVIEW

File: scripts/services/magazine.service.js [hunk 1]
```diff
@@ -1,44 +1,14 @@
...
-
-    return Array.isArray(data) ? data : [data];
-  } catch (error) {
-    // eslint-disable-next-line no-console
-    console.error(`Error fetching articles: ${error.message || error}`);
-    return [];
-  }
-};
+import { getLocale } from '../common.js';
+import { fetchData, magazineSearchQuery, TENANT } from '../search-api.js';
```
Comment (alexiscoelho) [inline scripts/services/magazine.service.js:2]: I know it is not on you but,
What do you think about renaming `fetchData` to `fetchSearchData`?
Seems to generic for me while it is a specific thing...

**Review by alexiscoelho — APPROVED** [review]

File: blocks/v2-article-search/v2-article-search.js [hunk 2]
```diff
@@ -62,6 +62,7 @@ const getTopics = async (props = {}) => {
       ...new Set(articles.items.map((item) => ({ key: 'category', value: item.value.trim() }))),
     ].sort((a, b) => a.value.localeCompare(b.value));
   } catch (error) {
+    // eslint-disable-next-line no-console
```
[the remark this answers is not in the document]
Comment (amglebihan) [reply blocks/v2-article-search/v2-article-search.js:65]: I know, it's just that seeing all the warnings in the terminal while running the linter made me to add the comments, OCD kicking in! :p I think it's fine, though. We should mostly avoid using comments for anything else, like we used to have a lot in the CSS and in some cases in the JS.

File: scripts/services/magazine.service.js (hunk 1 above)
Comment (amglebihan) [reply scripts/services/magazine.service.js]: Changed.

File: blocks/v2-stories-carousel/v2-stories-carousel.js [hunk 3]
```diff
@@ -95,75 +97,88 @@ const navigate = (carousel, direction, arrowLeftButton, arrowRightButton) => {
...
-  rightArrowListItem.append(arrowRightButton);
-
-  arrowControls.append(leftArrowListItem, rightArrowListItem);
-  carousel.insertAdjacentElement('beforebegin', arrowControls);
-
-  decorateIcons(arrowControls);
-
+  const arrowLeftButtonLabel = getTextLabel('Previous');
+  const arrowRightButtonLabel = getTextLabel('Next');
+  const arrowControlsContainer = document.createElement('ul');
```
[the remark this answers is not in the document]
Comment (amglebihan) [reply blocks/v2-stories-carousel/v2-stories-carousel.js]: Done.

File: blocks/related-articles/related-articles.js [hunk 4]
```diff
@@ -64,11 +88,11 @@ async function createRelatedtMagazineArticles(mainEl, magazineArticles) {
     const articleCard = buildRelatedMagazineArticle(entry);
     articleCards.appendChild(articleCard);
   });
-}
+};
 
 export default async function decorate(block) {
-  const allArticles = await fetchMagazineArticles();
+  const allArticles = await fetchMagazineArticles({ limit: 100 });
```
[the remark this answers is not in the document]
Comment (amglebihan) [reply blocks/related-articles/related-articles.js]: I have removed this limit in this block, and also used `sort: 'LAST_MODIFIED_DESC'`, as the block only displays three articles.

**Pull request merged**